sdk/sdktypes/user_id.go

```diff
@@ -0,0 +1,54 @@
+package sdktypes
+
+import (
+	"crypto/sha256"
+	"fmt"
+
+	"github.com/google/uuid"
+	"go.autokitteh.dev/autokitteh/internal/kittehs"
+)
+
+const userIDKind = "usr"
+
+type UserID = id[userIDTraits]
+
+type userIDTraits struct{}
+
+func (userIDTraits) Prefix() string { return userIDKind }
+
+// func NewUserID() UserID                          { return newID[UserID]() }
+func ParseUserID(s string) (UserID, error)       { return ParseID[UserID](s) }
+func StrictParseUserID(s string) (UserID, error) { return Strict(ParseUserID(s)) }
+
+var InvalidUserID UserID
+
+// FIXME: use provider ID
+func NewUserIDFromUserData(provider string, email string, name string) UserID {
```

Comment:
```suggestion
func NewUserIDFromUser(user User) UserID {
```

Reply:
see User.ID which reads the data and passes it to this method